File: test_login_errors.py
import requests
import json

BASE_URL = "http://localhost:5000/api/auth/login"

def test_login(email, password, description):
    print(f"Testing: {description}")
    payload = {
        "email": email,
        "password": password
    }
    headers = {
        "Content-Type": "application/json"
    }
    
    try:
        response = requests.post(BASE_URL, json=payload, headers=headers)
        print(f"Status Code: {response.status_code}")
        try:
            print(f"Response Body: {json.dumps(response.json(), indent=2)}")
        except:
            print(f"Response Text: {response.text}")
        print("-" * 30)
    except Exception as e:
        print(f"Error: {e}")
        print("-" * 30)

# 1. Test Invalid Email
test_login("nonexistent@example.com", "password123", "Invalid Email")

# 2. Test Invalid Password (need a valid user first, let's try to find one or assume one exists)
# If we don't know a valid user, we might only see the "Invalid Email" case. 
# But let's try a potentially valid email if we saw one in the file, or just rely on the first test to confirm the "Invalid email" message change.
# I will try to use 'admin@example.com' as a guess, or I can check the seed file if available.
# Actually, I can just create a test user first if needed, but for now let's see if the first test works.

File: test_login_errors.js
const BASE_URL = "http://localhost:5000/api/auth/login";

async function testLogin(email, password, description) {
    console.log(`Testing: ${description}`);
    try {
        const response = await fetch(BASE_URL, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ email, password })
        });

        console.log(`Status Code: ${response.status}`);
        const data = await response.json();
        console.log("Response Body:", JSON.stringify(data, null, 2));
    } catch (error) {
        console.error("Error:", error.message);
    }
    console.log("-".repeat(30));
}

// 1. Test Invalid Email
testLogin("nonexistent@example.com", "password123", "Invalid Email");

// 2. Test Invalid Password (create a user first if needed, but for now we test invalid email)
// To test invalid password, I need a valid user email. I'll try to register one first to be sure.

async function registerAndTestWeakPassword() {
    console.log("Registering a temporary user for password test...");
    const registerUrl = "http://localhost:5000/api/auth/register";
    const tempUser = {
        email: "test_temp_user_login_error@example.com",
        password: "ValidPassword123!",
        firstName: "Temp",
        lastName: "User"
    };

    try {
        // Register
        const regResponse = await fetch(registerUrl, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(tempUser)
        });

        if (regResponse.status === 201 || regResponse.status === 409) {
            console.log("User registered or already exists. Proceeding to test invalid password.");
            // Test Invalid Password
            await testLogin(tempUser.email, "WrongPassword123!", "Invalid Password");
        } else {
            console.log("Failed to register temp user:", regResponse.status);
            const data = await regResponse.json();
            console.log(data);
        }
    } catch (error) {
        console.error("Registration error:", error);
    }
}

registerAndTestWeakPassword();
